API for affaldsportal

TermeHansen opened the pull request.

TermeHansen commented:
@briis could you test this a bit on some of the renoweb municipalities?

briis commented:
I tested now with 4 different Renoweb municipalities. They all work, but for Hjørring kommune I got these errors:

````
WARNING:pyaffalddk.api:Material type [240 l plast/metal/drikkekartoner og papir/pap 4-ugers] is not defined in the system for Genbrug. Please notify the developer. Municipality: hjørring, Address ID: 24102
WARNING:pyaffalddk.api:Material type [240 l mad/restaffald 2 delt 14-dags] is not defined in the system for Genbrug. Please notify the developer. Municipality: hjørring, Address ID: 24102
````
So they should be added to the material list

briis commented:
Btw in relation to this issue with Hillerød:
> [@rhgiter](https://github.com/rhgiter) Jeg tror jeg ved fejlen. Hvis du kigger på adressen i Perfect Waste appen en gang virker det nok i HA med version 2.4.0 bagefter. Jeg ser om jeg ikke kan fikse til næste version...

> Tak, den kom ind da jeg brugte appen.

I picked a random address in Hillerød, and had no issues getting data

TermeHansen commented:
Jeg ved hvad det er. Perfect Waste har en intern database for grupper af veje ser det ud til. Hvis den ikke har været aktiveret før skal man sende et ekstra kald.

TermeHansen commented:
fixed missing material types and also added the fix for the missing pickup data in the perfect waste interface.

I would like to make some more tests and make the code cleaner for this with getting correct fraction names, but that work should be in a separate PR.

briis commented:
I re-tested with the same Municipalities as before and they all work, and the missing data from Hjørring are fixed.
I now tried with an extra Municipality: Albertslund, they are on the supported list, but keep returning this error:
`Error 500 from https://europe-west3-perfect-waste.cloudfunctions.net/searchExternalAddresses`
I have tried with these random adresses, which all look to be Parcelhuse:
- enebærhaven 61
- tranehusene 35
- borgager 22

TermeHansen commented:
I'll look into it...

TermeHansen commented:
Do we know the data should work for them? in Perfect Waste app there is no data, and changing to affaldsportal api I also see no data for Albertslund ?!?

briis commented:
No, I don't know. They were just 3 random addresses.
I looked in the issues log for affalddk and found [this issue](https://github.com/briis/affalddk/issues/129) that relates to Albertslund. The addressid is 695, but that throws some other errors. And I don't know if that is a correct id anymore

briis commented:
Now that you are close to a new update, did you also see this new issue with missing data for Ringkøbing/Skjern kommune:
https://github.com/briis/affalddk/issues/266

TermeHansen commented:
They should be there now with this I believe...

TermeHansen commented:
I suggest we release this, so we can test it in HA for a new pre-release. This with Albertslund we can take in a separate issue after this release. As I don't see any data in the Perfect Waste app it could easy be something on their side of things.

briis commented:
Agree. Thanks.

TermeHansen commented:
@briis will you draft a new pre-release for the HA addon to have people test, and hopefully close some of the issues there...

briis commented:
Yes. I am not home right now, but will be in 1-2 hours. Then I will make the new pre-release

briis commented:
I am working on the release right now. With first support for Affaldsportalen, what other Municipalities are now supported, do we know?
